node/resolvers/Queries/index.ts

```diff
@@ -102,12 +115,17 @@ const Index = {
     const variables = {
       ...(organizationId && { organizationId }),
       ...(costCenterId && { costCenterId }),
+      page,
+      pageSize,
+      search,
+      sortOrder,
+      sortedBy,
     }
 
     return storefrontPermissions
       .listUsers(variables)
       .then((result: any) => {
-        return result.data.listUsers
+        return result.data.listUsersPaginated
       })
```

Comment:
The permissions check in this resolver needs to be updated to allow sales admins to view users in any organization. You'll notice that if the `adminUserAuthToken` is present, all of the permissions checks are skipped. Since you are an admin user, that's what's allowing the query to return all users in your workspace. 

Similar to what you did in the Cost Centers query, you should check the user's role. Then you could change the conditional to `if (!adminUserAuthToken && !isSalesAdmin) { ...` . Otherwise the sales admin will only be able to view users that are part of the sales admin's "main" organization.